contracts/Reputation.sol

```diff
@@ -20,8 +20,10 @@ contract Reputation is Ownable {
         return reputation[_owner];
     }
 
-	function set_reputation(address _account, uint256 _amount) onlyOwner {
+	function setReputation(address _account, uint256 _amount) onlyOwner {
+	    uint prevTotalReputation = totalReputation; 
         totalReputation = totalReputation - reputation[_account] + _amount;
+        if( ( _amount >= reputation[_account] ) && ( prevTotalReputation > totalReputation ) ) throw; // overflow 
```

Comment:
You are right. I though about that. But then it is probably best to change the entire codebase to use safe math.